src/domain/AOReward.ts

```diff
@@ -62,17 +63,39 @@ export function calculateAORewards(date: BigInt, pool: Pool): void {
   systemRewards.save()
 }
 
-function setAORewardRate(systemRewards: RewardDayTotal): RewardDayTotal {
-  log.debug('setting system AO rewards rate, toDateAORewardAggregateValue {}', [
-    systemRewards.toDateAORewardAggregateValue.toString(),
-  ])
-  if (systemRewards.toDateAORewardAggregateValue.lt(BigDecimal.fromString(aoRewardsCeiling))) {
-    systemRewards.aoRewardRate = BigDecimal.fromString('0.0017')
-    systemRewards.save()
+function getAORewardRate(systemRewards: RewardDayTotal): BigDecimal {
+  let defaultRewardRate = BigDecimal.fromString('0.0017')
+
+  let isBelowLimit = systemRewards.toDateRewardAggregateValue.lt(BigDecimal.fromString(aoRewardsCeiling))
```

Comment:
Did you try logging this value? Could well be that there is something wrong with decimals here.